### PR TITLE
Update pre-commit config

### PR DESCRIPTION
**Pull request by plaplant**
## Description

For security purposes, GitHub has discontinued support for `git://` URLs. We had one such URL in our pre-commit yaml file. This should fix the errors in CI.

## Motivation and Context

I also took this opportunity to update the pre-commit dependencies, which led to some style formatting changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [x] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Build or continuous integration change checklist:
- [ ] If required or optional dependencies have changed (including version numbers), I have updated the readme to reflect this.
- [ ] If this is a new CI setup, I have added the associated badge to the readme and to references/make_index.py (if appropriate).

### CODE REVIEW

File: pyuvdata/uvdata/uvdata.py [hunk 1]
```diff
@@ -8143,8 +8208,26 @@ def _select_preprocess(
...
+                + [
+                    isinstance(
+                        item[1],
+                        (
+                            int,
+                            np.integer,
+                        ),
+                    )
+                    for item in bls
+                ]
```
Comment (bhazelton) [inline pyuvdata/uvdata/uvdata.py:8230]: I think this is too hard to read and needs to be refactored in some way to make it better.

File: pyuvdata/uvdata/uvdata.py [hunk 2]
```diff
@@ -8126,7 +8184,14 @@ def _select_preprocess(
...
             if isinstance(bls, list) and all(
-                isinstance(bl_ind, (int, np.integer,),) for bl_ind in bls
+                isinstance(
+                    bl_ind,
+                    (
+                        int,
+                        np.integer,
+                    ),
+                )
+                for bl_ind in bls
```
Comment (bhazelton) [inline pyuvdata/uvdata/uvdata.py:8194]: This is also a bit ugly.

File: pyuvdata/uvflag/uvflag.py [hunk 3]
```diff
@@ -2074,8 +2077,26 @@ def _select_preprocess(
...
+                + [
+                    isinstance(
+                        item[1],
+                        (
+                            int,
+                            np.integer,
+                        ),
+                    )
+                    for item in bls
+                ]
```
Comment (bhazelton) [inline pyuvdata/uvflag/uvflag.py:2099]: Another candidate for refactoring. I find this painful to try to parse.

**Review by bhazelton — COMMENTED** [review]
This is great, but Black has gotten more aggressive with carriage returns and is making some parts of the code very hard to read. I think we should do some kind of refactoring to address this -- maybe defining new variables to keep the logic on one line.

**Comment by bhazelton** [comment]
I'm fine with taking out the linter action in favor of the pre-commit service.

File: pyuvdata/uvflag/uvflag.py (hunk 3 above)
Comment (steven-murray) [reply pyuvdata/uvflag/uvflag.py:2099]: Current versions of `black` will not expand this as long as it fits on one line and the arguments don't end with a trailing comma.

File: pyuvdata/uvdata/tests/test_aipy_extracts.py [hunk 4]
```diff
@@ -343,7 +344,17 @@ def test_add_to_header(tmp_path):
...
+            -1,
+            "(0x,1x)_(2y,3y)",
+            (
+                0,
+                1,
+                2,
+                3,
+            ),
+            76,
+        ],
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply pyuvdata/uvdata/tests/test_aipy_extracts.py:357]: I believe it's the trailing comma in the tuple

File: pyuvdata/uvflag/uvflag.py (hunk 3 above)
Comment (bhazelton) [reply pyuvdata/uvflag/uvflag.py:2099]: I think the change is caused by an upgrade to Black, from 19.10b0 to 22.1.0.

File: pyuvdata/uvdata/tests/test_aipy_extracts.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (plaplant) [reply pyuvdata/uvdata/tests/test_aipy_extracts.py]: Fixed in b5ea2cbd.

File: pyuvdata/uvdata/uvdata.py (hunk 1 above)
Comment (plaplant) [reply pyuvdata/uvdata/uvdata.py]: Fixed in b5ea2cbd.

File: pyuvdata/uvdata/uvdata.py (hunk 2 above)
Comment (plaplant) [reply pyuvdata/uvdata/uvdata.py]: Fixed in b5ea2cbd.

File: pyuvdata/uvflag/uvflag.py (hunk 3 above)
Comment (plaplant) [reply pyuvdata/uvflag/uvflag.py]: Fixed in b5ea2cbd.

**Review by bhazelton — DISMISSED** [review]
Looks good, thanks for cleaning those things up.

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -16,7 +16,7 @@ repos:
           - id: mixed-line-ending
             args: ['--fix=no']
     -   repo: https://gitlab.com/pycqa/flake8
-        rev: '3.7.9'
+        rev: '3.9.2'
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply .pre-commit-config.yaml:19]: I'm looking to do this same thing on pyradiosky and we're already using 4.0.1 there.

[the remark this answers is not in the document]
Comment (bhazelton) [reply .pre-commit-config.yaml:19]: Oh! we also have a different URL: https://github.com/pycqa/flake8

Which I believe is the currently recommended one.

**Review by bhazelton — DISMISSED** [review]

**Review by bhazelton — APPROVED** [review]

**Comment by bhazelton** [comment]
The hera_qm test is failing because it tries to install hera_cal, which has dependencies specified in it's setup.py with urls that start with "git://". This doesn't break the hera_cal tests because we install with `--no-deps`.

I made an issue about this on hera_cal here: https://github.com/HERA-Team/hera_cal/issues/766

**Comment by bhazelton** [comment]
The hera_cal issue was fixed, so I re-ran the hera_qm job.

**Pull request merged**